src/build.cc

```diff
@@ -557,12 +557,12 @@ bool Builder::AddTarget(Node* target, string* err) {
     return false;
 
   if (Edge* in_edge = target->in_edge()) {
-    if (in_edge->outputs_ready())
-      return true;  // Nothing to do.
+    if (!in_edge->outputs_ready() &&
+        !plan_.AddTarget(target, err)) {
```

Comment:
Yes, it causes a slight change in behavior if you attempt to build a leaf node that doesn't exist, printing `no work to do` instead of the previous `ninja: error: 'a' missing and no known rule to make it`.

Reply:
Done.